ansible/roles-infra/infra-osp-dns/tasks/instance_loop.yml

```diff
@@ -9,10 +9,15 @@
     _instance_name: "{{ _instance.name }}"
 
 - when: _instance.count|int > 1
+  vars:
+      # Be very conservative, ensure every records are deleted
+      # regardless of if it was before or after the patch that makes the instances
+      # start at index1 and not index0.
+      _start_at: "{{ 0 if _dns_state == 'absent' else 1 }}"
   block:
-    - loop: "{{ range(1, _instance.count | int + 1) | list }}"
+    - loop: "{{ range(_start_at | int, _instance.count | int + 1) | list }}"
```

Comment:
Can we name these tasks? Really all of the tasks in this task list should be updated with names to adhere to our standards. Non-named tasks make JK sad.